IdentityCore/tests/automation/MSIDClientCredentialHelper.m

```diff
@@ -0,0 +1,249 @@
+
+    NSMutableURLRequest *request = [[NSMutableURLRequest alloc] initWithURL:[NSURL URLWithString:tokenEndpoint]];
+    [request setHTTPMethod:@"POST"];
+    [request setValue:@"application/x-www-form-urlencoded" forHTTPHeaderField:@"Content-Type"];
+
+    NSData *requestBody = [[postParams msidURLFormEncode] dataUsingEncoding:NSUTF8StringEncoding];
+    [request setHTTPBody:requestBody];
+
+    NSURLSession *session = [NSURLSession sessionWithConfiguration:[NSURLSessionConfiguration defaultSessionConfiguration]];
+
+    [[session dataTaskWithRequest:request
+                completionHandler:^(NSData * _Nullable data,
+                                    NSURLResponse * _Nullable response, NSError * _Nullable error)
+      {
+          if (error)
+          {
+              if (completionHandler)
+              {
+                  completionHandler(nil, error);
+              }
+              return;
+          }
+
+          NSDictionary *result = [NSJSONSerialization JSONObjectWithData:data options:0 error:nil];
+
+          NSError *msidError = nil;
+          MSIDAADV1TokenResponse *tokenResponse = [[MSIDAADV1TokenResponse alloc] initWithJSONDictionary:result error:&msidError];
+
+          if (msidError)
+          {
+              if (completionHandler)
+              {
+                  completionHandler(nil, msidError);
+              }
+
+              return;
+          }
+
+          MSIDRequestParameters *params = [MSIDRequestParameters new];
+          params.authority = [NSURL URLWithString:authority];
+          params.clientId = clientId;
+          params.target = resource;
+
+          MSIDAADV1Oauth2Strategy *strategy = [MSIDAADV1Oauth2Strategy new];
+
+          BOOL checkResult = [strategy verifyResponse:tokenResponse context:nil error:&msidError];
+
+          if (!checkResult)
+          {
+              if (completionHandler)
+              {
+                  completionHandler(nil, msidError);
+              }
+          }
+
+          MSIDAccessToken *accessToken = [strategy accessTokenFromResponse:tokenResponse request:params];
+          self.accessTokenCache[cacheKey] = accessToken;
+
+          if (completionHandler)
+          {
+              completionHandler(accessToken.accessToken, nil);
+          }
+
+      }] resume];
+
+}
+
++ (NSString *)clientCertificateAssertionForAudience:(NSString *)audience
+                                           clientId:(NSString *)clientID
+                                    certificateData:(NSData *)certificateData
+                                           password:(NSString *)password
+{
+    SecIdentityRef identity = [self createIdentityFromData:certificateData password:password];
+
+    if (!identity)
+    {
+        NSLog(@"Couldn't load identity!");
+        return nil;
+    }
+
+    SecKeyRef privateKey = nil;
+    OSStatus result = SecIdentityCopyPrivateKey(identity, &privateKey);
+
+    if (result != errSecSuccess || !privateKey)
```

Comment:
nit: I'd reorder the conditionals here for better readability

Reply:
What's your preference?

Reply:
remove pkey!!!

Reply:
done